client/apis/volumegroupsnapshot/v1alpha1/types.go

```diff
@@ -0,0 +1,284 @@
+	// recreated with different set of values, and as such, should not be referenced
+	// post-snapshot creation.
+	// For dynamic provisioning, this field must be set.
+	// This field may be unset for pre-provisioned snapshots.
+	// +optional
+	VolumeGroupSnapshotClassName *string `json:"volumeGroupSnapshotClassName,omitempty" protobuf:"bytes,4,opt,name=volumeGroupSnapshotClassName"`
+
+	// Source specifies whether the snapshot is (or should be) dynamically provisioned
+	// or already exists, and just requires a Kubernetes object representation.
+	// This field is immutable after creation.
+	// Required.
+	Source VolumeGroupSnapshotContentSource `json:"source" protobuf:"bytes,5,opt,name=source"`
+}
+
+// VolumeGroupSnapshotContentStatus defines the observed state of VolumeGroupSnapshotContent.
+type VolumeGroupSnapshotContentStatus struct {
+	// VolumeGroupSnapshotHandle is a unique id returned by the CSI driver
+	// to identify the VolumeGroupSnapshot on the storage system.
+	// If a storage system does not provide such an id, the
+	// CSI driver can choose to return the VolumeGroupSnapshot name.
+	// +optional
+	VolumeGroupSnapshotHandle *string `json:"volumeGroupSnapshotHandle,omitempty" protobuf:"bytes,1,opt,name=volumeGroupSnapshotHandle"`
+
+	// CreationTime is the timestamp when the point-in-time group snapshot is taken
+	// by the underlying storage system.
+	// If not specified, it indicates the creation time is unknown.
+	// +optional
+	CreationTime *int64 `json:"creationTime,omitempty" protobuf:"varint,2,opt,name=creationTime"`
+
+	// ReadyToUse indicates if all the individual snapshots in the group are ready to be
+	// used to restore a volume.
+	// If not specified, it means the readiness of a group snapshot is unknown.
+	// +optional
+	ReadyToUse *bool `json:"readyToUse,omitempty" protobuf:"varint,3,opt,name=readyToUse"`
+
+	// Error is the last observed error during group snapshot creation, if any.
+	// Upon success after retry, this error field will be cleared.
+	// +optional
+	Error *snapshotv1.VolumeSnapshotError `json:"error,omitempty" protobuf:"bytes,4,opt,name=error,casttype=VolumeSnapshotError"`
+
+	// VolumeSnapshotContentRefList is the list of volume snapshot content references
+	// for this group snapshot.
+	// The maximum number of allowed snapshots in the group is 100.
+	// +optional
+	VolumeSnapshotContentRefList []core_v1.ObjectReference `json:"volumeSnapshotRefList,omitempty" protobuf:"bytes,5,opt,name=volumeSnapshotRefList"`
+}
+
+// VolumeGroupSnapshotContentSource represents the CSI source of a group snapshot.
+// Exactly one of its members must be set.
+// Members in VolumeGroupSnapshotContentSource are immutable.
+type VolumeGroupSnapshotContentSource struct {
+	// PersistentVolumeNames is a list of names of PersistentVolumes to be snapshotted
+	// together. Signifies dynamic provisioning of the VolumeGroupSnapshot.
+	// This field is immutable.
```

Comment:
Signifies dynamic provisioning of the VolumeGroupSnapshot. -> It specified for dynamic provisioning of the VolumeGroupSnapshot.